shell/platform/linux/fl_renderer.cc

```diff
@@ -271,125 +217,29 @@ gboolean fl_renderer_present_layers(FlRenderer* self,
   return TRUE;
 }
 
-void fl_renderer_setup(FlRenderer* self) {
-  FlRendererPrivate* priv = reinterpret_cast<FlRendererPrivate*>(
-      fl_renderer_get_instance_private(self));
-
-  g_return_if_fail(FL_IS_RENDERER(self));
-
-  GLuint vertex_shader = glCreateShader(GL_VERTEX_SHADER);
-  glShaderSource(vertex_shader, 1, &vertex_shader_src, nullptr);
-  glCompileShader(vertex_shader);
-  int vertex_compile_status;
-  glGetShaderiv(vertex_shader, GL_COMPILE_STATUS, &vertex_compile_status);
-  if (vertex_compile_status == GL_FALSE) {
-    g_autofree gchar* shader_log = get_shader_log(vertex_shader);
-    g_warning("Failed to compile vertex shader: %s", shader_log);
-  }
-
-  GLuint fragment_shader = glCreateShader(GL_FRAGMENT_SHADER);
-  glShaderSource(fragment_shader, 1, &fragment_shader_src, nullptr);
-  glCompileShader(fragment_shader);
-  int fragment_compile_status;
-  glGetShaderiv(fragment_shader, GL_COMPILE_STATUS, &fragment_compile_status);
-  if (fragment_compile_status == GL_FALSE) {
-    g_autofree gchar* shader_log = get_shader_log(fragment_shader);
-    g_warning("Failed to compile fragment shader: %s", shader_log);
-  }
-
-  priv->program = glCreateProgram();
-  glAttachShader(priv->program, vertex_shader);
-  glAttachShader(priv->program, fragment_shader);
-  glLinkProgram(priv->program);
-
-  int link_status;
-  glGetProgramiv(priv->program, GL_LINK_STATUS, &link_status);
-  if (link_status == GL_FALSE) {
-    g_autofree gchar* program_log = get_program_log(priv->program);
-    g_warning("Failed to link program: %s", program_log);
-  }
-
-  glDeleteShader(vertex_shader);
-  glDeleteShader(fragment_shader);
-}
-
 void fl_renderer_render(FlRenderer* self, int width, int height) {
   FlRendererPrivate* priv = reinterpret_cast<FlRendererPrivate*>(
       fl_renderer_get_instance_private(self));
 
   g_return_if_fail(FL_IS_RENDERER(self));
 
-  // Save bindings that are set by this function.  All bindings must be restored
-  // to their original values because Skia expects that its bindings have not
-  // been altered.
-  GLint saved_texture_binding;
-  glGetIntegerv(GL_TEXTURE_BINDING_2D, &saved_texture_binding);
-  GLint saved_vao_binding;
-  glGetIntegerv(GL_VERTEX_ARRAY_BINDING, &saved_vao_binding);
-  GLint saved_array_buffer_binding;
-  glGetIntegerv(GL_ARRAY_BUFFER_BINDING, &saved_array_buffer_binding);
-
   glClearColor(0.0, 0.0, 0.0, 1.0);
   glClear(GL_COLOR_BUFFER_BIT);
```

Comment:
FYI, you'll probably want to disable the stencil test here. See this comment by @bdero: https://github.com/flutter/engine/pull/49726#discussion_r1451038236

Corresponding Windows logic:

https://github.com/flutter/engine/blob/a338661d909c51dc15aedcba0978a80ec3a3c234/shell/platform/windows/compositor_opengl.cc#L142-L160